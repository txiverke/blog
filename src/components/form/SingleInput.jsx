// @flow

import React from 'react'

type Props = {
  wrapper?: string,
  name: string,
  inputType: mixed,
  title: string,
  controlFunc?: Function,
  content?: string,
  placeholder?: string,
  size?: string,
  required?: boolean,
  pattern?: string,
}

const SingleInput = (props: Props) => (
  <div className={props.wrapper}>
    <label
      id={props.name + 'Label'}
      htmlFor={props.name} 
      className="app-form-label"
    >
      {props.title}
    </label>
    <input
      id={props.name + 'Input'}
      className="app-form-input"
      name={props.name}
      type={props.inputType}
      autoComplete={props.name}
      defaultValue={props.content}
      onChange={props.controlFunc}
      placeholder={props.placeholder}
      pattern={props.pattern}
      required={props.required}
    />
    <p className="app-form-error" id={props.name + 'Error'}></p>
  </div>
) 
  
SingleInput.defaultProps = {
  wrapper: 'app-form-group',
  content: '',
  placeholder: '',
  size: '',
  pattern: '',
  required: true,
  controlFunc: () => {}
}

export default SingleInput