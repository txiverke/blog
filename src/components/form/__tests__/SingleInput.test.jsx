import React from 'react'
import SingleInput from '../SingleInput'
import { shallow } from 'enzyme'

describe('<SingleInput />', () => {
  it('should render itself', () => {
    const name = 'test_name'
    const inputType = 'text'
    const title = 'Test name'
    const component = shallow(<SingleInput name={name} inputType={inputType} title={title} />)
    expect(component).toMatchSnapshot()
    expect(component.find('.app-form-group').length).toEqual(1)
    expect(component.text()).toEqual(title)
  })

  it('should overwrite the default props', () => {
    const name = 'test_name'
    const inputType = 'text'
    const title = 'Test name'
    const wrapper = 'test-wrapper'
    const component = shallow(<SingleInput name={name} inputType={inputType} title={title} wrapper={wrapper} />)
    expect(component).toMatchSnapshot()
    expect(component.find('.app-form-group').length).toEqual(0)
    expect(component.find('.test-wrapper').length).toEqual(1)
  })
})