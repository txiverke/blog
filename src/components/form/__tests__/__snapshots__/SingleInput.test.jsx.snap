// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleInput /> should overwrite the default props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SingleInput
    content=""
    controlFunc={[Function]}
    inputType="text"
    name="test_name"
    pattern=""
    placeholder=""
    required={true}
    size=""
    title="Test name"
    wrapper="test-wrapper"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className="app-form-label"
          htmlFor="test_name"
          id="test_nameLabel"
>
          Test name
</label>,
        <input
          autoComplete="test_name"
          className="app-form-input"
          defaultValue=""
          id="test_nameInput"
          name="test_name"
          onChange={[Function]}
          pattern=""
          placeholder=""
          required={true}
          type="text"
/>,
        <p
          className="app-form-error"
          id="test_nameError"
/>,
      ],
      "className": "test-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test name",
          "className": "app-form-label",
          "htmlFor": "test_name",
          "id": "test_nameLabel",
        },
        "ref": null,
        "rendered": "Test name",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoComplete": "test_name",
          "className": "app-form-input",
          "defaultValue": "",
          "id": "test_nameInput",
          "name": "test_name",
          "onChange": [Function],
          "pattern": "",
          "placeholder": "",
          "required": true,
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "app-form-error",
          "id": "test_nameError",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="app-form-label"
            htmlFor="test_name"
            id="test_nameLabel"
>
            Test name
</label>,
          <input
            autoComplete="test_name"
            className="app-form-input"
            defaultValue=""
            id="test_nameInput"
            name="test_name"
            onChange={[Function]}
            pattern=""
            placeholder=""
            required={true}
            type="text"
/>,
          <p
            className="app-form-error"
            id="test_nameError"
/>,
        ],
        "className": "test-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test name",
            "className": "app-form-label",
            "htmlFor": "test_name",
            "id": "test_nameLabel",
          },
          "ref": null,
          "rendered": "Test name",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoComplete": "test_name",
            "className": "app-form-input",
            "defaultValue": "",
            "id": "test_nameInput",
            "name": "test_name",
            "onChange": [Function],
            "pattern": "",
            "placeholder": "",
            "required": true,
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "app-form-error",
            "id": "test_nameError",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<SingleInput /> should render itself 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SingleInput
    content=""
    controlFunc={[Function]}
    inputType="text"
    name="test_name"
    pattern=""
    placeholder=""
    required={true}
    size=""
    title="Test name"
    wrapper="app-form-group"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className="app-form-label"
          htmlFor="test_name"
          id="test_nameLabel"
>
          Test name
</label>,
        <input
          autoComplete="test_name"
          className="app-form-input"
          defaultValue=""
          id="test_nameInput"
          name="test_name"
          onChange={[Function]}
          pattern=""
          placeholder=""
          required={true}
          type="text"
/>,
        <p
          className="app-form-error"
          id="test_nameError"
/>,
      ],
      "className": "app-form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test name",
          "className": "app-form-label",
          "htmlFor": "test_name",
          "id": "test_nameLabel",
        },
        "ref": null,
        "rendered": "Test name",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoComplete": "test_name",
          "className": "app-form-input",
          "defaultValue": "",
          "id": "test_nameInput",
          "name": "test_name",
          "onChange": [Function],
          "pattern": "",
          "placeholder": "",
          "required": true,
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "app-form-error",
          "id": "test_nameError",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="app-form-label"
            htmlFor="test_name"
            id="test_nameLabel"
>
            Test name
</label>,
          <input
            autoComplete="test_name"
            className="app-form-input"
            defaultValue=""
            id="test_nameInput"
            name="test_name"
            onChange={[Function]}
            pattern=""
            placeholder=""
            required={true}
            type="text"
/>,
          <p
            className="app-form-error"
            id="test_nameError"
/>,
        ],
        "className": "app-form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test name",
            "className": "app-form-label",
            "htmlFor": "test_name",
            "id": "test_nameLabel",
          },
          "ref": null,
          "rendered": "Test name",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoComplete": "test_name",
            "className": "app-form-input",
            "defaultValue": "",
            "id": "test_nameInput",
            "name": "test_name",
            "onChange": [Function],
            "pattern": "",
            "placeholder": "",
            "required": true,
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "app-form-error",
            "id": "test_nameError",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
